src/core/core/rpc_message.cpp

```diff
@@ -415,14 +499,24 @@ message_ex* message_ex::create_receive_message(const blob& data)
     return msg;
 }
 
-message_ex* message_ex::create_receive_message_with_standalone_header(const blob& data)
+message_ex* message_ex::create_receive_message_with_standalone_header(const blob& data, bool header_included)
```

Comment:
The above ```create_receive_message``` means header is included.  Here we allocate the buffer for message header because header is always not included. I.e., we already separate these two scenarios into two different methods.